### PR TITLE
use indicies instead of phantom proofs

### PR DESCRIPTION
**Pull request by ShapeOfMatter**

### CODE REVIEW

**Comment by ShapeOfMatter** [comment]
This isolates one of the big refactors of #52: it removes `gdp` as a dependency and upgrades the proof objects from phantoms to pattern-matchable indices. 

Included in this is refactoring `Faceted` as wrappers around functions, but without bringing in `forLocs` we can't actually _build_ them that way.

File: src/Choreography/Internal/Location.hs [hunk 1]
```diff
@@ -28,77 +25,71 @@ wrap = Wrap
...
-instance {-# OVERLAPS #-} ExplicitMember x (x ': xs) where
-  explicitMember = axiom
+unsafeFacet :: [Maybe a] -> Member l ls -> Located '[l] a
+unsafeFacet (Just a : _) First = Wrap a
+unsafeFacet (Nothing : _) First = Empty
+unsafeFacet (_ : as) (Later l) = unsafeFacet as l
+unsafeFacet [] _ = error "The provided list isn't long enough to use as a Faceted over the intended parties."
+
+
+data Member (x :: k) (xs :: [k]) where
```
Comment (psilospore) [inline src/Choreography/Internal/Location.hs:42]: This is nice! What do you think about some documentation or examples? It took me a bit to understand this after being away from this project for a while:

```suggestion
{- | A proof that a location is a member of a list of locations.
 __Examples:__

@
aliceIsFirst = First :: Member "alice" '["alice", "bob"]
bobIsSecond = Later First ::  Member "bob" '[ "alice", "bob"]
@
-}
data Member (x :: k) (xs :: [k]) where
```

Comment (psilospore) [reply src/Choreography/Internal/Location.hs:42]: I can do the others if you think this would be useful

Comment (ShapeOfMatter) [reply src/Choreography/Internal/Location.hs:42]: Documentation is obviously lacking and good. I think the example you have above is over-verbose; might be better to just say "The structure of a `Member l ls` encodes a natural/Peano number, but the signatures of the constructors enforce that said number is an index into `ls` at which `l` appears".

**Pull request merged**